complex objects/cargo/list/src/lib.rs

```diff
@@ -195,14 +197,18 @@ impl ListManager {
                 match item_iter.next() {
                     Some(result) => {
                         match result {
-                            Ok(item) => Some(item),
+                            Ok(item) => {
+                                println!("Returning created item {:?}", item);
+                                Some(item)
```

Comment:
FYI, `Result::ok()` turns a `Result` into an `Option`. 

Digression into ways to handle this kind of thing: we added some [utilities in Mentat](https://github.com/mozilla/mentat/blob/20aa11dcbdabd03b0274b849f640caff84b46882/core/src/util.rs#L11) to chain side-effects off both `Result` and `Option`, so you could replace this whole `match` block with something flatter, like:

```
item_iter.next()
         .when_ok((item) => { println!("Returning created item {:?}", item); })
         .when_err((e) => { println!("Failed to fetch item {:?}", e); })
         .ok()
```

and then just kill those `when` lines when you no longer want the debugging.